redfish-core/lib/redfish_sessions.hpp

```diff
@@ -140,84 +140,134 @@ inline void requestRoutesSession(App& app)
+                if (!json_util::readJson(*oemObject, asyncResp->res, "OpenBMC",
+                                         bmcOem))
+                {
                     return;
                 }
-
-                int pamrc = pamAuthenticateUser(username, password);
-                bool isConfigureSelfOnly = pamrc == PAM_NEW_AUTHTOK_REQD;
-                if ((pamrc != PAM_SUCCESS) && !isConfigureSelfOnly)
+                if (!json_util::readJson(*bmcOem, asyncResp->res, "ClientID",
+                                         clientId))
                 {
-                    messages::resourceAtUriUnauthorized(
-                        asyncResp->res, std::string(req.url),
-                        "Invalid username or password");
+                    BMCWEB_LOG_ERROR << "Could not read ClientId";
                     return;
                 }
-#ifdef BMCWEB_ENABLE_IBM_MANAGEMENT_CONSOLE
-                if (oemObject)
-                {
-                    std::optional<nlohmann::json> bmcOem;
-                    if (!json_util::readJson(*oemObject, asyncResp->res,
-                                             "OpenBMC", bmcOem))
+            }
+#endif
+            // Check the user info before creating session
+            crow::connections::systemBus->async_method_call(
+                [&req, asyncResp, username, password, clientId,
+                 isConfigureSelfOnly](
+                    const boost::system::error_code ec,
+                    std::map<std::string,
+                             std::variant<bool, std::string,
+                                          std::vector<std::string>>>
+                        userInfo) {
+                    if (ec)
                     {
+                        BMCWEB_LOG_ERROR << "GetUserInfo failed...";
+                        asyncResp->res.result(
+                            boost::beast::http::status::internal_server_error);
                         return;
                     }
-                    if (!json_util::readJson(*bmcOem, asyncResp->res,
-                                             "ClientID", clientId))
+
+                    const std::string* userRolePtr = nullptr;
+                    auto userInfoIter = userInfo.find("UserPrivilege");
+                    if (userInfoIter != userInfo.end())
                     {
-                        BMCWEB_LOG_ERROR << "Could not read ClientId";
-                        return;
+                        userRolePtr =
+                            std::get_if<std::string>(&userInfoIter->second);
+                    }
+                    std::string userRole{};
+                    if (userRolePtr != nullptr)
+                    {
+                        userRole = *userRolePtr;
+                        BMCWEB_LOG_DEBUG << "userName = " << username
+                                         << " userRole = " << *userRolePtr;
+                        ::redfish::Privileges userPrivileges =
```

Comment:
I have never seen it like this, ::redfish, can we have these match, e.g.  redfish::privileges::getSessionService

Reply:
Done